src/test/java/io/vertx/test/fakedns/DnsMessageEncoder.java

```diff
@@ -0,0 +1,233 @@
+package io.vertx.test.fakedns;
+
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+
+  import java.io.IOException;
```

Comment:
why do we need to fork this class ?

Reply:
I explained this in the commit message where this was done: https://github.com/eclipse-vertx/vert.x/pull/4908/commits/50d7d82964b17436bdee94fce312dd69493bee44 Plus also noted this in the PR description.

TL;DR: `DnsMessageEncoder` does not know how to encode `DNAME` records at the moment,. So in order to reproduce the bug using `FakeDNSServer`, the `DnsMessageEncoder` class needs to be extended.

Unfortutely the design of `DnsMessageEncoder` does not lend itself to extension. The `DEFAULT_ENCODERS` map is final and uses `Collections.unmodifiableMap`, and the method that is used to look up the encoder from `DEFAULT_ENCODERS` has `private` access modifier, so doing a subclass wouldn't work either.